mlflow/java/client/src/main/java/org/mlflow/tracking/MlflowClient.java

```diff
@@ -167,26 +219,79 @@ public RunInfo createRun(CreateRun request) {
    */
-  public List<Run> searchRuns(List<String> experimentIds,
+  public RunsPage searchRuns(List<String> experimentIds,
                                   String searchFilter,
                                   ViewType runViewType,
                                   List<String> orderBy) {
+    return searchRuns(experimentIds, searchFilter, runViewType, orderBy, 1000);
+  }
+
+  /**
+   * Return runs from provided list of experiments that satisfy the search query.
+   *
+   * @param experimentIds List of experiment IDs.
+   * @param searchFilter SQL compatible search query string. Format of this query string is
+   *                     similar to that specified on MLflow UI.
+   *                     Example : "params.model = 'LogisticRegression' and metrics.acc != 0.9"
+   * @param runViewType ViewType for expected runs. One of (ACTIVE_ONLY, DELETED_ONLY, ALL)
+   *                    Defaults to ACTIVE_ONLY.
+   * @param orderBy List of properties to order by. Example: "metrics.acc DESC".
+   * @param maxResults Maximum number of runs desired. Defaults to 1000.
+   *
+   * @return A list of all Runs that satisfy search filter.
+   */
+  public RunsPage searchRuns(List<String> experimentIds,
+                              String searchFilter,
+                              ViewType runViewType,
+                              List<String> orderBy,
+                              int maxResults) {
+    return searchRuns(experimentIds,searchFilter, runViewType, orderBy, maxResults,null);
+  }
+
+  /**
+   * Return runs from provided list of experiments that satisfy the search query.
+   *
+   * @param experimentIds List of experiment IDs.
+   * @param searchFilter SQL compatible search query string. Format of this query string is
+   *                     similar to that specified on MLflow UI.
+   *                     Example : "params.model = 'LogisticRegression' and metrics.acc != 0.9"
+   * @param runViewType ViewType for expected runs. One of (ACTIVE_ONLY, DELETED_ONLY, ALL)
+   *                    Defaults to ACTIVE_ONLY.
+   * @param orderBy List of properties to order by. Example: "metrics.acc DESC".
+   * @param maxResults Maximum number of runs desired. Defaults to 1000.
+   * @param pageToken String token specifying the next page of results. It should be obtained from
+   *             a call to {@link #searchRuns(List, String)}.
+   *
+   * @return A page of Runs that satisfy the search filter.
+   */
+  public RunsPage searchRuns(List<String> experimentIds,
+                              String searchFilter,
+                              ViewType runViewType,
+                              List<String> orderBy,
+                              int maxResults,
+                              String pageToken) {
     SearchRuns.Builder builder = SearchRuns.newBuilder()
-      .addAllExperimentIds(experimentIds)
-      .addAllOrderBy(orderBy);
+            .addAllExperimentIds(experimentIds)
```

Comment:
can users send in `null` for `experimentIds` -- if we don't want that can we add `@NotNull` or handle that in the method like done in the other `searchRuns` method that calls REST endpoint. ditto for `orderBy` argument.

Reply:
I believe @NotNull is just a doc annotation. I don't think it actually adds any validation of the input. I'm not sure what would happen if we didn't check (not sure what protobuf's generated code does), but I can try it out. I believe the expected behavior should be to throw an exception.

Reply:
Yup, it throws a `NullPointerException`